src/main/java/Duke/Duke.java

```diff
@@ -0,0 +1,284 @@
+            }
+        }
+    }
+
+    static void processCommand(String userInput) throws DukeException {
+        String command = userInput.split(" ")[0];
+
+        switch (command) {
+        case "todo":
+            addTodo(userInput);
+            break;
+        case "deadline":
+            addDeadline(userInput);
+            break;
+        case "event":
+            addEvent(userInput);
+            break;
+        case "done":
+            setTaskToBeDone(userInput);
+            break;
+        case "bye":
+            exit();
+            break;
+        case "list":
+            displayList();
+            break;
+        case "delete":
+            deleteTask(userInput);
+            break;
+        default:
+            throw new DukeException(WRONG_COMMAND_ERROR_MESSAGE);
+        }
+    }
+
+    public static void addTask(Task task) {
+        tasks.add(numberOfTasks, task);
+        numberOfTasks++;
+        addTask = true;
+        echo(task, addTask);
+    }
+
+    static void addTodo(String userInput) throws DukeException {
+        String todo = userInput.substring(4);
+        todo = todo.replace(" ", "");
+
+        if (todo.isEmpty()) {
+            throw new DukeException(MISSING_TODO_CONTENT_MESSAGE);
+        } else {
+            Task newTask = new Todo(userInput.substring(5));
+            addTask(newTask);
+        }
+        try {
+            writeToFile("duke.txt", tasks);
+        } catch (IOException e) {
+            System.out.println("File not found");
+        }
+    }
+
+    static void addDeadline(String userInput) throws DukeException {
+        String deadline = userInput.substring((8));
+        deadline = deadline.replace(" ", "");
+
+        if (deadline.isEmpty()) {
+            throw new DukeException(MISSING_DEADLINE_CONTENT_MESSAGE);
+        } else {
+            Task newTask = new Deadline(userInput.substring(9, userInput.indexOf("/")),
+                    userInput.substring(userInput.indexOf("/by") + 4));
+            addTask(newTask);
+        }
+        try {
+            writeToFile("duke.txt", tasks);
+        } catch (IOException e) {
+            System.out.println("File not found");
+        }
+    }
+
+    static void addEvent(String userInput) throws DukeException {
+        String event = userInput.substring((5));
+        event = event.replace(" ", "");
+
+        if (event.isEmpty()) {
+            throw new DukeException(MISSING_EVENT_CONTENT_MESSAGE);
+        } else {
+            Task newTask = new Event(userInput.substring(6, userInput.indexOf("/")),
+                    userInput.substring(userInput.indexOf("/at") + 4));
+            addTask(newTask);
+        }
+        try {
+            writeToFile("duke.txt", tasks);
+        } catch (IOException e) {
+            System.out.println("File not found");
+        }
+    }
+
+    static void setTaskToBeDone(String userInput) {
+        int selectedTask = Integer.parseInt(userInput.substring(5)) - 1;
```

Comment:
what do the numbers mean?

Reply:
Sorry! I will avoid magic numbers by refactoring my code to improve readability and level of maintenance.